pkg/occlient/occlient.go

```diff
@@ -1572,46 +1478,169 @@ func (c *Client) GetOnePodFromSelector(selector string) (*corev1.Pod, error) {
 	return &pods.Items[0], nil
 }
 
-// RsyncPath copies local directory to directory in running Pod.
-func (c *Client) RsyncPath(localPath string, targetPodName string, targetPath string) (string, error) {
-	log.Debugf("Syncing %s to pod %s:%s", localPath, targetPodName, targetPath)
+// CopyFile copies single local file to the directory in running Pod.
+// if asFile is true, localFile will be file otherwise it is path to a directory
+// watchedFiles is list of changed files captured during `odo watch`
+func (c *Client) CopyFile(asFile bool, localFile string, targetPodName string, targetPath string, watchedFiles []string) error {
```

Comment:
can we  again rename `watchedFiles` here to something better? Like `files` or `filesToPush` or `copyFiles` There is no watching happening here. Variables should be named according to their meaning in the function not how they are called outside of the function.

There should be also a commend explaining that `localFile` is ignored if `copyFiles` is set.